code/src/layouts/dashboard-layout/DashboardLayout.tsx

```diff
@@ -1,37 +1,23 @@
-import { Suspense } from "react";
-import { Outlet } from "react-router-dom";
+import DashboardAdminLayout from "@/layouts/dashboard-layout/components/dashboard-admin-tab-layout/DashboardAdminLayout";
+import DashboardManagerLayout from "@/layouts/dashboard-layout/components/dashboard-manager-tab-layout/DashboardManagerLayout";
 
-import { dashboardTabs } from "@/layouts/dashboard-layout/DashboardLayout.constants";
-import DashboardTab from "@/layouts/dashboard-layout/components/dashboard-tab/DashboardTab";
-import PageWrapper from "@/layouts/page-wrapper/PageWrapper";
+import { ROLES } from "@/constants/common";
+import { useUserRoleSelector } from "@/store/slices/userSlice";
 
-import PageLoadingFallback from "@/containers/page-loading-fallback/PageLoadingFallback";
+import "@/layouts/dashboard-layout/components/dashboard-tab-layout/DashboardTabLayout.scss";
 
-import AppBox from "@/components/app-box/AppBox";
+const DashboardLayout = () => {
+  const userRole = useUserRoleSelector();
 
-import "@/layouts/dashboard-layout/DashboardLayout.scss";
+  if (userRole === ROLES.ADMIN) {
+    return <DashboardAdminLayout />;
+  }
 
-const DashboardLayout = () => {
-  const tabLabels = dashboardTabs.map((tab) => (
-    <DashboardTab key={tab.name} tab={tab} />
-  ));
+  if (userRole === ROLES.SHOP_MANAGER) {
+    return <DashboardManagerLayout />;
+  }
```

Comment:
I guess you can use object here and use layouts by Role as a key

Reply:
Thanks! I used `switch case`! I discussed it with you